thinc/config.py

```diff
@@ -195,10 +196,24 @@ def copy(self) -> "Config":
             raise ValueError(f"Couldn't deep-copy config: {e}")
         return Config(config)
 
-    def from_str(self, text: str) -> "Config":
-        "Load the config from a string."
+    def _set_overrides(self, config: "ConfigParser", overrides: Dict[str, Any]) -> None:
+        """Set overrides in the ConfigParser before config is interpreted."""
+        err_title = "Error parsing config overrides"
+        for key, value in overrides.items():
+            parts = key.split(".")
```

Comment:
This is not really significant, and maybe you find this code structure worse. But how about:

```
if "." not in key:
    raise ConfigValidationError("", err, message=err_title)
section, field = key.rsplit(".", 1)
if section not in config:
    raise ConfigValidationError("", err, message=err_title) 
config.set(section, field, srsly.json_dumps(value))
```